### PR TITLE
Revert "[dynamo] Graph break on `setattr(Tensor, "data", Tensor)` (#113043)"

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #113297

This reverts commit ddfe5725342b0c0f707222879ca9dac305f97210.

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

@diff-train-skip-merge

### CODE REVIEW

File: test/dynamo/test_input_attr_tracking.py [hunk 1]
```diff
@@ -369,3 +369,8 @@ def do_some_setattr_stuff(self):
         # call_function  add     <built-in function add>  (l_x_, l_x_)          {}
         # call_function  mul_1   <built-in function mul>  (mul, add)            {}
         # output         output  output                   ((mul_1, mul, add),)  {}
+
+if __name__ == "__main__":
+    from torch._dynamo.test_case import run_tests
+
+    run_tests()
```
Comment (voznesenskym) [inline test/dynamo/test_input_attr_tracking.py:376]: I added this - these tests were failing internal but not running on CI

**Review by PaliC — APPROVED** [review]

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D51132884).

**Comment by huydhn** [comment]
For posterity, I'm going to delete `@pytorchbot revert` message from `facebook-github-bot` to see if it help fix ShipIt.

**Comment by huydhn** [comment]
Also, deleting comment from mergebot because the PR has been merged

**Pull request closed without merging**